src/Core/Grid/Data/Factory/OrderStatusesGridDataFactoryDecorator.php

```diff
@@ -0,0 +1,93 @@
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Image\ImageProviderInterface;
+
+/**
+ * Class OrderStatusesGridDataFactoryDecorator decorates data from order_statuses doctrine data factory.
+ */
+final class OrderStatusesGridDataFactoryDecorator implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $orderStatusesDoctrineGridDataFactory;
+    /**
+     * @var ImageProviderInterface
+     */
+    private $orderStatusIconProvider;
+
+    /**
+     * @param GridDataFactoryInterface $orderStatutesDoctrineGridDataFactory
+     */
+    public function __construct(
+        GridDataFactoryInterface $orderStatutesDoctrineGridDataFactory,
+        ImageProviderInterface $orderStatusIconProvider
+    ) {
+        $this->orderStatusesDoctrineGridDataFactory = $orderStatutesDoctrineGridDataFactory;
+        $this->orderStatusIconProvider = $orderStatusIconProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $orderStatutesData = $this->orderStatusesDoctrineGridDataFactory->getData($searchCriteria);
+
+        $orderStatutesRecords = $this->applyModifications($orderStatutesData->getRecords());
+
+        return new GridData(
+            $orderStatutesRecords,
+            $orderStatutesData->getRecordsTotal(),
+            $orderStatutesData->getQuery()
+        );
+    }
+
+    /**
+     * @param RecordCollectionInterface $orderStatuses
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $orderStatuses)
+    {
+        $modifiedOrderStatuses = [];
+
+        foreach ($orderStatuses as $orderStatus) {
```

Comment:
```suggestion
        foreach ($orderStatuses as &$orderStatus) {
```
Instead of populate a new array :thinking: 
Or with array_map or something similare

Reply:
removed the icon field so this method is not necessary anymore